### DISCUSSION TITLE
DietPi-Software | Netmaker

### DISCUSSION THREAD

**Discussion by HyperCriSiS** [MichaIng/DietPi#5500] [Software Request]
Netmaker is a tool for creating and managing virtual overlay networks. If you have at least two machines with internet access that you need to connect with a secure tunnel, Netmaker is for you. If you have thousands of servers spread across multiple locations, data centers, or clouds, Netmaker is also for you. Netmaker connects machines securely, wherever they are.

Netmaker has many similarities to Tailscale, ZeroTier, and Nebula. What makes Netmaker different is the speed and flexibility. Netmaker is faster because it uses kernel WireGuard. It is more dynamic because the server and agents are fully configurable, which lets you handle all sorts of different use cases. And, of course, you can also self-host Netmaker, which gives you complete control of your network traffic.

https://www.netmaker.io
https://github.com/gravitl/netmaker

![waterfox_3N42UqHx2f](https://github.com/MichaIng/DietPi/assets/11708192/9182937c-3ed4-452b-8756-376317a78988)

Vote for this software implementation with the ⬆️ button on this post.

**Comment by HyperCriSiS** [comment]
https://medium.com/netmaker/battle-of-the-vpns-which-one-is-fastest-speed-test-21ddc9cd50db
![grafik](https://user-images.githubusercontent.com/11708192/176012096-1b15f362-5ab5-45be-929e-e03bcbd6047b.png)

**Reply by HyperCriSiS** [reply]
The huge advantage from Netmaker is, that no special client is needed. Just a wireguard client is needed to connect to the Netmaker server.

**Comment by MichaIng** [comment]
Quite interesting. "Real" WireGuard without the need to open a port, right? We could share some of the WireGuard code responsible to check or compile the WireGuard kernel module.

**Reply by HyperCriSiS** [reply]
The server seems to need 443, 53, and 51821-51830/udp open.

**Reply by Joulinar** [reply]
according this install guide, it is not needed to open port 53 https://netmaker.readthedocs.io/en/master/quick-start.html#open-firewall
Question is, if this is required on local firewall only, or on the router port forwarding as well.

**Reply by MichaIng** [reply]
Jep, it needs `51821` + 1 for each network and `443` for admin panel, API etc.
